src/metabase/util/encryption.clj

```diff
@@ -147,22 +147,22 @@
   "If `MB_ENCRYPTION_SECRET_KEY` is set and `v` is encrypted, decrypt `v`; otherwise return `s` as-is. Attempts to check
   whether `v` is an encrypted String, in which case the decrypted String is returned, or whether `v` is encrypted bytes,
   in which case the decrypted bytes are returned."
-  {:arglists '([secret-key? s & {:keys [log-errors?], :or {log-errors? true}}])}
+  {:arglists '([secret-key? s & [{:keys [log-errors?], :or {log-errors? true}}]])}
   [& args]
-  (let [[secret-key & more]        (if (and (bytes? (first args)) (string? (second args))) ;TODO: fix hackiness
+  ;; secret-key as an argument so that tests can pass it directly without using `with-redefs` to run in parallel
+  (let [[secret-key v options]     (if (and (bytes? (first args)) (string? (second args)))
                                      args
                                      (cons default-secret-key args))
-        [v & options]              more
         {:keys [log-errors?]
-         :or   {log-errors? true}} (apply hash-map options)
+         :or   {log-errors? true}} options
         log-error-fn (fn [kind ^Throwable e]
                        (when log-errors?
                          (log/warn (trs "Cannot decrypt encrypted {0}. Have you changed or forgot to set MB_ENCRYPTION_SECRET_KEY?"
```

Comment:
Isn't it sufficient to change this from `log/warn` to `log/error`?

I tried this and the test passes
```
diff --git a/src/metabase/models/interface.clj b/src/metabase/models/interface.clj
index 891c2e4362..a644be3977 100644
--- a/src/metabase/models/interface.clj
+++ b/src/metabase/models/interface.clj
@@ -234,7 +234,7 @@
       (catch Throwable e
         (if (or (encryption/possibly-encrypted-string? decrypted)
                 (encryption/possibly-encrypted-bytes? decrypted))
-          (log/error e "Could not decrypt encrypted field! Have you forgot to set MB_ENCRYPTION_SECRET_KEY?")
+          nil #_(log/error e "Could not decrypt encrypted field! Have you forgot to set MB_ENCRYPTION_SECRET_KEY?")
           (log/error e "Error parsing JSON"))  ; same message as in `json-out`
         v))))
 
diff --git a/test/metabase/models/interface_test.clj b/test/metabase/models/interface_test.clj
index 5e2b59b22e..55275f2c5b 100644
--- a/test/metabase/models/interface_test.clj
+++ b/test/metabase/models/interface_test.clj
@@ -145,8 +145,8 @@
              (mi/encrypted-json-out
               (encryption/encrypt (encryption/secret-key->hash "qwe") "{\"a\": 1}"))))))
   (testing "Logs an error message when incoming data looks encrypted"
-    (is (=? [[:error JsonParseException "Could not decrypt encrypted field! Have you forgot to set MB_ENCRYPTION_SECRET_KEY?"]]
-            (tu.log/with-log-messages-for-level :error
+    (is (=? [[:warn nil #"^Cannot decrypt encrypted String\. Have you changed or forgot to set MB_ENCRYPTION_SECRET_KEY\? Message seems corrupt or manipulated\.[\s\S]*"]]
+            (tu.log/with-log-messages-for-level :warn
               (mi/encrypted-json-out
                (encryption/encrypt (encryption/secret-key->hash "qwe") "{\"a\": 1}"))))))
```

Reply:
Not really, it gets into `(nil? secret-key)` branch and never does anything. The problem here is not that key is wrong, it's that it's not present.